Dockerfile

```diff
@@ -1,93 +1,46 @@
+
+RUN add-apt-repository ppa:deadsnakes/ppa
+RUN apt-get install --no-install-recommends -y python3.11-dev
+RUN curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+RUN apt-get install --no-install-recommends -y libjsoncpp-dev libspdlog-dev pigz
+RUN python3.11 get-pip.py
+
+RUN curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
+RUN add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+RUN apt-get update
+RUN apt-get install --no-install-recommends -y docker-ce-cli
+
+RUN git clone --single-branch --depth 1 https://github.com/mfontanini/cppkafka.git
+
+WORKDIR /home/ubuntu/software
+RUN git clone --single-branch --depth 1 --branch v5.1.1-DistDGL-v0.5 https://github.com/KarypisLab/METIS.git
+WORKDIR /home/ubuntu/software/METIS
+RUN git submodule update --init
+RUN make config shared=1 cc=gcc prefix=/usr/local
+RUN make install
+
+RUN mkdir /home/ubuntu/software/cppkafka/build
 WORKDIR /home/ubuntu/software/cppkafka/build
-RUN cmake .. && \
-    make && \
-    make install
-WORKDIR /home/ubuntu/software
-RUN wget https://archive.apache.org/dist/xerces/c/3/sources/xerces-c-3.2.2.tar.gz  && \
-    tar -xvf xerces-c-3.2.2.tar.gz
-WORKDIR /home/ubuntu/software/xerces-c-3.2.2
-RUN sh configure --disable-transcoder-icu && \
-    make install
-WORKDIR /home/ubuntu/software
-RUN git clone https://github.com/open-source-parsers/jsoncpp.git
-WORKDIR /home/ubuntu/software/jsoncpp
-RUN git checkout tags/1.8.4 && \
-    mkdir -p build/debug
-WORKDIR /home/ubuntu/software/jsoncpp/build/debug
-RUN cmake -DCMAKE_BUILD_TYPE=debug -DBUILD_STATIC_LIBS=ON -DBUILD_SHARED_LIBS=OFF -DARCHIVE_INSTALL_DIR=. -G "Unix Makefiles" ../.. && \
-    make
-WORKDIR /home/ubuntu/software
-RUN wget http://zlib.net/pigz/pigz-2.6.tar.gz
-RUN tar -xvf pigz-2.6.tar.gz
-WORKDIR /home/ubuntu/software/pigz-2.6
-RUN make
-ENV PATH="/home/ubuntu/software/pigz-2.6/pigz:${PATH}"
-WORKDIR /home/ubuntu/software
-RUN wget https://nchc.dl.sourceforge.net/project/nmon/nmon16m_helpsystems.tar.gz
-RUN mkdir nmon
-RUN tar -xf nmon16m_helpsystems.tar.gz -C nmon
-WORKDIR /home/ubuntu/software/nmon
-RUN cp nmon_x86_64_ubuntu18 /usr/local/bin
-WORKDIR /home/ubuntu/software
-RUN mkdir nlohmann_json
-WORKDIR /home/ubuntu/software/nlohmann_json
-RUN wget https://github.com/nlohmann/json/releases/download/v3.9.1/include.zip
-RUN unzip include.zip
-WORKDIR /home/ubuntu/software
-RUN apt-get update && \
-    apt-get install -y software-properties-common && \
-    apt-get update && \
-    add-apt-repository ppa:deadsnakes/ppa && \
-    apt-get install -y python3.11-dev && \
-    apt-get install -y libtool && \
-    git clone https://github.com/DamithaSenevirathne/jasminegraph-prerequisites.git
-ENV HOME="/home/ubuntu"
-WORKDIR /home/ubuntu/software/jasminegraph-prerequisites/python_libs
-RUN apt-get install -y python-pip && \
-    apt-get update && \
-    pip install -r requirements
```

Comment:
Installing GraphSAGE requirements (python libraries) is removed. Is it okay as the plan to get rid of GraphSAGE?